src/runtime/TemporalObject.cpp

```diff
@@ -0,0 +1,562 @@
+    : Temporal(state, proto)
+{
+}
+
+Value TemporalObject::toISODateTime(ExecutionState& state, DateObject& d)
+{
+    auto result = std::to_string(d.getFullYear(state))
+        + "-"
+        + (d.getMonth(state) + 1 < 10 ? "0" : "")
+        + std::to_string(d.getMonth(state) + 1)
+        + "-"
+        + (d.getDate(state) < 10 ? "0" : "")
+        + std::to_string(d.getDate(state))
+        + "T"
+        + (d.getHours(state) < 10 ? "0" : "")
+        + std::to_string(d.getHours(state))
+        + ":"
+        + (d.getMinutes(state) < 10 ? "0" : "")
+        + std::to_string(d.getMinutes(state))
+        + ":"
+        + (d.getSeconds(state) < 10 ? "0" : "")
+        + std::to_string(d.getSeconds(state)) + "+00:00";
+    return Value(new ASCIIString(result.data(), result.length()));
+}
+
+Value TemporalObject::toISODate(ExecutionState& state, DateObject& d)
+{
+    auto result = std::to_string(d.getFullYear(state)) + "-" + (d.getMonth(state) + 1 < 10 ? "0" : "") + std::to_string(d.getMonth(state) + 1) + "-" + (d.getDate(state) < 10 ? "0" : "") + std::to_string(d.getDate(state));
+    return Value(new ASCIIString(result.data(), result.length()));
+}
+
+Value TemporalObject::toISOTime(ExecutionState& state, DateObject& d)
+{
+    auto result = (d.getHours(state) < 10 ? "0" : "") + std::to_string(d.getHours(state)) + ":" + (d.getMinutes(state) < 10 ? "0" : "") + std::to_string(d.getMinutes(state)) + ":" + (d.getSeconds(state) < 10 ? "0" : "") + std::to_string(d.getSeconds(state)) + "." + std::to_string(d.getMilliseconds(state));
+    return Value(new ASCIIString(result.data(), result.length()));
+}
+
+std::map<std::string, Value> TemporalObject::parseValidIso8601String(ExecutionState& state, const Value& str)
+{
+    ASSERT(str.isString());
+    size_t pos = str.asString()->toNonGCUTF8StringData().find('[');
+    std::string isoDate = str.asString()->toNonGCUTF8StringData().substr(0, pos);
+    std::string extra;
+
+    if (pos != std::string::npos) {
+        extra = str.asString()->toNonGCUTF8StringData().substr(pos);
+    }
+
+    pos = isoDate.find('T');
+
+    if (pos != std::string::npos) {
+        std::replace(isoDate.begin(), isoDate.end(), 'T', ' ');
+    }
+
+    pos = isoDate.find(' ');
+    std::string date = str.asString()->toNonGCUTF8StringData().substr(0, pos);
+    std::string time = str.asString()->toNonGCUTF8StringData().substr(pos + 1);
+    Value year;
+    Value month;
+    Value week;
+    Value day;
+    pos = date.find('-');
+
+    if (pos != std::string::npos) {
+        if (pos == 0 && date.length() > 4) {
+            if (date.substr(pos + 1).find('-') == 0) {
+                date.erase(std::remove(date.begin(), date.end(), '-'), date.end());
+                if (date.length() == 4) {
+                    month = Value(Value(date.substr(1, 2).c_str()).asInt32());
+                    day = Value(Value(date.substr(3, 2).c_str()).asInt32());
+                    year = Value("");
```

Comment:
Why did you wrap values with `Value` here?
How about directly calculating strings with primitive types?